benchmarks/bench_rcv1_logreg_convergence.py

```diff
@@ -0,0 +1,234 @@
+ccat_idx = rcv1.target_names.tolist().index('CCAT')
+y = rcv1.target.tocsc()[:, ccat_idx].toarray().ravel().astype(np.float64)
+y[y == 0] = -1
+
+# parameters
+C = 1.
+fit_intercept = True
+tol = 1.0e-14
+
+# max_iter range
+sgd_iter_range = list(range(1, 121, 10))
+newton_iter_range = list(range(1, 25, 3))
+lbfgs_iter_range = list(range(1, 242, 12))
+liblinear_iter_range = list(range(1, 37, 3))
+liblinear_dual_iter_range = list(range(1, 85, 6))
+sag_iter_range = list(range(1, 37, 3))
+
+clfs = [
+    ("LR-liblinear",
+     LogisticRegression(C=C, tol=tol,
+                        solver="liblinear", fit_intercept=fit_intercept,
+                        intercept_scaling=1),
+     liblinear_iter_range, [], [], [], []),
+    ("LR-liblinear-dual",
+     LogisticRegression(C=C, tol=tol, dual=True,
+                        solver="liblinear", fit_intercept=fit_intercept,
+                        intercept_scaling=1),
+     liblinear_dual_iter_range, [], [], [], []),
+    ("LR-SAG",
+     LogisticRegression(C=C, tol=tol,
+                        solver="sag", fit_intercept=fit_intercept),
+     sag_iter_range, [], [], [], []),
+    ("LR-newton-cg",
+     LogisticRegression(C=C, tol=tol, solver="newton-cg",
+                        fit_intercept=fit_intercept),
+     newton_iter_range, [], [], [], []),
+    ("LR-lbfgs",
+     LogisticRegression(C=C, tol=tol,
+                        solver="lbfgs", fit_intercept=fit_intercept),
+     lbfgs_iter_range, [], [], [], []),
+    ("SGD",
+     SGDClassifier(alpha=1.0 / C / n_samples, penalty='l2', loss='log',
+                   fit_intercept=fit_intercept, verbose=0),
+     sgd_iter_range, [], [], [], [])]
+
+
+# if lightning_clf is not None and not fit_intercept:
+#     # compute the same eta than in LR-sag
+#     random_state = check_random_state(1)
+#     sample_weight = np.ones(n_samples, dtype=np.float64, order='C')
+#     X_dataset, intercept_decay = make_dataset(X, y.astype(np.float64),
+#                                               sample_weight, random_state)
+#     eta = get_auto_eta(X_dataset, 1. / C / n_samples, n_samples, Log(), 0)
+
+#     clfs.append(
+#         ("Lightning-SVRG",
+#          lightning_clf.SVRGClassifier(alpha=1. / C / n_samples, eta=eta,
+#                                       tol=tol, loss="log"),
+#          sag_iter_range, [], [], [], []))
+#     clfs.append(
+#         ("Lightning-SAG",
+#          lightning_clf.SAGClassifier(alpha=1. / C / n_samples, eta=eta,
+#                                      tol=tol, loss="log"),
+#          sag_iter_range, [], [], [], []))
+
+#     # We keep only 200 features, to have a dense dataset,
+#     # and compare to lightning SAG, which seems incorrect in the sparse case.
+#     X_csc = X.tocsc()
+#     nnz_in_each_features = X_csc.indptr[1:] - X_csc.indptr[:-1]
+#     X = X_csc[:, np.argsort(nnz_in_each_features)[-200:]]
+#     X = X.toarray()
+#     print("dataset: %.3f MB" % (X.nbytes / 1e6))
```

Comment:
I ran the benchmark with `fit_intercept=False` and full sparse dataset, and forgot to uncomment it.
I will